apps/meteor/app/livechat/server/api/v1/contact.js

```diff
@@ -51,19 +51,23 @@ API.v1.addRoute(
 				check(this.queryParams, {
 					email: Match.Maybe(String),
 					phone: Match.Maybe(String),
+					custom: Match.Maybe(String),
 				});
+				const { email, phone, custom } = this.queryParams;
 
-				const { email, phone } = this.queryParams;
-
-				if (!email && !phone) {
+				if (!email && !phone && !custom) {
 					throw new Meteor.Error('error-invalid-params');
 				}
-
+				const allowedCF = LivechatCustomField.find({ scope: 'visitor' }, { fields: { _id: 1 } }, { searchability: true }).map(
+					({ _id }) => _id,
+				);
+				const customObj = Object.fromEntries(Array.from(new URLSearchParams(custom)).filter(([k]) => allowedCF.includes(k)));
```

Comment:
Can you share an example of what `custom` would look like? Wanna see why we need to manually parsing the searchparams 👀

Reply:
sure, we can make one with new URLSearchParams({field:1,other:"string"}).toString(), for example, that leads to `field=1&other=string`, and the quoted line way to get the object to put on the query.

Reply:
it gets URL encoded, so it's end up as `custom=a%3D1%26b%3D1`, very ugly but works.